ios/RNLinksdk.m

```diff
@@ -113,44 +116,57 @@ - (void)stopObserving {
     PLKProduct product = PLKProductFromArray(products);
     // v1 is no longer supported, always use v2 as default.
     PLKAPIVersion apiVersion = kPLKAPIVersionDefault;
-    PLKConfiguration* linkConfiguration = [[PLKConfiguration alloc] initWithKey:key
-                                                                           env:environment
-                                                                       product:product
-                                                                 selectAccount:selectAccount
-                                                                  longtailAuth:longtailAuth
-                                                                    apiVersion:apiVersion];
-    if ([clientName length] > 0) {
-       linkConfiguration.clientName = clientName;
-    }
-    if([linkCustomizationName length] > 0) {
-        linkConfiguration.linkCustomizationName = linkCustomizationName;
-    }
-    if ([webhook length] > 0) {
-       linkConfiguration.webhook = [NSURL URLWithString:webhook];
-    }
-    if ([userLegalName length] > 0) {
-       linkConfiguration.userLegalName = userLegalName;
-    }
-    if ([userEmailAddress length] > 0) {
-       linkConfiguration.userEmailAddress = userEmailAddress;
-    }
-    if ([userPhoneNumber length] > 0) {
-        linkConfiguration.userPhoneNumber = userPhoneNumber;
-    }
-    if ([oauthRedirectUri length] > 0) {
-        linkConfiguration.oauthRedirectUri = [NSURL URLWithString:oauthRedirectUri];
-    }
-    if ([oauthNonce length] > 0) {
-        linkConfiguration.oauthNonce = oauthNonce;
-    }
-    if ([accountSubtypes count] > 0) {
-       linkConfiguration.accountSubtypes = accountSubtypes;
-    }
-    if ([countryCodes count] > 0) {
-       linkConfiguration.countryCodes = countryCodes;
-    }
-    if ([language length] > 0) {
-       linkConfiguration.language = language;
+    PLKConfiguration* linkConfiguration;
+    BOOL isUsingLinkToken = [linkTokenInput length] > 0 && [linkTokenInput hasPrefix:kRNLinkKitLinkTokenPrefix];
+
+    if (isUsingLinkToken) {
+      linkConfiguration = [[PLKConfiguration alloc] initWithLinkToken:linkTokenInput];
+      if ([oauthRedirectUri length] > 0) {
+          linkConfiguration.oauthRedirectUri = [NSURL URLWithString:oauthRedirectUri];
+      }
+      if ([oauthNonce length] > 0) {
+          linkConfiguration.oauthNonce = oauthNonce;
+      }
```

Comment:
💅 How about creating two methods, one for creating a `linkConfiguration` for use with link token and another one for creating a `linkConfiguration` as before.

Reply:
Good idea! ⚡